dbtools/pg.py

```diff
@@ -752,9 +779,50 @@ def get_table_column_details(self, table: str, schema: str) -> List[ColumnDetail
         detailed_columns = [ColumnDetails(*r) for r in results]
         return detailed_columns
 
+    def get_view_column_details(self, matview: str, schema: str) -> List[ColumnDetails]:

```

Comment:
The unit test that hits this function passes, and this may not be a concern, but I still wanted to flag:
I tried this query out and it returned columns with types (name, text, boolean, text, text), which are different than the types hinted in the `ColumnDetails` dataclass. Running the test `test_get_view_column_details`, it looks like the `ColumnDetails` instance `test_column` has `str`, `str`, `bool`, `None`, `None` typed attributes.

![2023-10-19_15-17](https://github.com/disbr007/dbtools/assets/68197166/2354e382-7c8f-4ff9-8876-f94ce81e32c8)

Reply:
Yeah so when you get the details of the matview from the database, those columns that are coming back as None are not available. I just made them `Optional` on the `ColumnDetails` class

Reply:
@disbr007 Makes sense. Last thing is I think the `is_nullable` column comes back as `str` when running `get_table_column_details` and `bool` when running `get_view_column_details`. Not sure if we'd want to annotate as `Union[bool, str]`?

Reply:
Good find! I just updated so that attribute is always a string. I'd have preferred it to be `bool` but the table column details method is more directly reading a pg system table, whereas the view details is already doing some transformations to try to fit that mold, so I stuck with what the table column method does.